modmesh/timeseries_dataframe.py

```diff
@@ -0,0 +1,105 @@
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import os
+import numpy as np
+
+from .core import SimpleArrayUint64, SimpleArrayFloat64
+
+
+__all__ = [
+    'TimeSeriesDataFrame'
+]
+
+
+class TimeSeriesDataFrame(object):
+
+    def __init__(self):
+
+        self._columns = list()
+        self._index_column = None
+        self._index_column_name = None
+        self._data = list()
+
+    def read_from_text_file(
+        self,
+        txt_path,
+        delimiter=',',
+        timestamp_in_file=True,
+        timestamp_column=None
+    ):
+        """
+        Generate dataframe from a text file.
+
+        :param txt_path: path to the text file.
+        :type txt_path: str
+        :param delimiter: delimiter.
+        :type delimiter: str
+        :param timestamp_in_file: If the text file containing index column,
+        expected to be integer.
+        :type timestamp_in_file: bool
+        :prarm timestamp_column: Column which stores timestamp data.
+        :type timestamp_column: str
+        :return None
+        """
+        if not os.path.exists(txt_path):
+            raise Exception("Text file '{}' does not exist".format(txt_path))
+
+        nd_arr = np.genfromtxt(txt_path, delimiter=delimiter)[1:]
+        index_column_num = 0 if timestamp_in_file else None
+
+        with open(txt_path, 'r') as f:
+            table_header = [x for x in f.readline().strip().split(delimiter)]
+            if timestamp_in_file:
+                if timestamp_column in table_header:
+                    index_column_num = table_header.index(timestamp_column)
+                self._index_column = SimpleArrayUint64(
```

Comment:
I wonder why it's `SimpleArrayUint64`? Can't it be a floating number?

Reply:
Timestamp should be an integer (like epoch), can you provide any other case that we need a floating point?

Reply:
I think it depends on the unit? for example, if the unit is ms, and the precision is ns, then the timestamp can be a floating number.

Reply:
Hi @tigercosmos 
The reason I designed in this way is because of the data from the model problem we are trying to use is integer, and it seems normal to treat timestamp an integer. I think the scenario you refer to is that the meaning of the time is the delta from certain starting point (1.2s, 2.0s, ...). This is not in our model problem so I will keep as is for this PR. But we can definitely open an issue when we encounter this kind of data.

Reply:
Sure, it's clearer to me.